src/utils/circuit-to-svg.tsx

```diff
@@ -0,0 +1,317 @@
+function createGeometryFromPolygons(polygons: any[]) {
+  const geometry = new BufferGeometry()
+  const vertices: number[] = []
+  const normals: number[] = []
+
+  try {
+    for (const polygon of polygons) {
+      // Create triangles from polygon vertices
+      for (let i = 1; i < polygon.vertices.length - 1; i++) {
+        vertices.push(
+          ...polygon.vertices[0],    // First vertex
+          ...polygon.vertices[i],    // Second vertex
+          ...polygon.vertices[i + 1] // Third vertex
+        )
+
+        // Add normal for each vertex of the triangle
+        const v1 = new THREE.Vector3(...polygon.vertices[0])
+        const v2 = new THREE.Vector3(...polygon.vertices[i])
+        const v3 = new THREE.Vector3(...polygon.vertices[i + 1])
+        const normal = new THREE.Vector3()
+          .crossVectors(
+            new THREE.Vector3().subVectors(v2, v1),
+            new THREE.Vector3().subVectors(v3, v1)
+          )
+          .normalize()
+        
+        normals.push(
+          normal.x, normal.y, normal.z,
+          normal.x, normal.y, normal.z,
+          normal.x, normal.y, normal.z
+        )
+      }
+    }
+
+    geometry.setAttribute('position', new Float32BufferAttribute(vertices, 3))
+    geometry.setAttribute('normal', new Float32BufferAttribute(normals, 3))
+    return geometry
+  } catch (error) {
+    console.error('Error creating geometry:', error)
+    throw error
+  }
+}
+
+async function loadModel(url: string): Promise<THREE.Object3D | null> {
+  try {
+    if (url.endsWith('.stl')) {
+      const loader = new STLLoader()
+      const geometry = await loader.loadAsync(url)
+      const material = new THREE.MeshStandardMaterial({
+        color: 0x888888,
+        metalness: 0.5,
+        roughness: 0.5
+      })
+      return new THREE.Mesh(geometry, material)
+    } else if (url.endsWith('.obj')) {
+      const loader = new OBJLoader()
+      return await loader.loadAsync(url)
+    }
+    return null
+  } catch (error) {
+    console.error('Failed to load model:', error)
+    return null
+  }
+}
+
+export async function circuitToSvg(
+  circuitJson: AnySoupElement[],
+  options: CircuitToSvgOptions = {}
+): Promise<string> {
+  const {
+    width = 800,
+    height = 600,
+    viewAngle = "top",
+    backgroundColor = "#ffffff",
+    padding = 20,
+    zoom = 1.5
+  } = options
+
+  try {
+    // Initialize scene and renderer
+    const scene = new THREE.Scene()
+    const renderer = new SVGRenderer()
+    renderer.setSize(width, height)
+    renderer.setClearColor(backgroundColor)
+
+    // Modify camera setup for better component visibility
+    const camera = new THREE.OrthographicCamera(
+      width / -2 / zoom,
+      width / 2 / zoom,
+      height / 2 / zoom,
+      height / -2 / zoom,
+      -1000,  // Changed near plane to negative to see below board
+      1000
+    )
+    camera.position.set(0, 0, 100)  // Move camera further out
+    camera.up.set(0, 0, 1)
+    camera.lookAt(0, 0, 0)
```

Comment:
let's make the default camera similar to the default camera in the current 3d viewer, we can have `cameraPosition` as a parameter to customize it